app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/CustomUserRepositoryImpl.java

```diff
@@ -25,8 +26,15 @@ public CustomUserRepositoryImpl(ReactiveMongoOperations mongoOperations, MongoCo
     @Override
     public Mono<User> findByEmail(String email, AclPermission aclPermission) {
         Criteria emailCriteria = where(fieldName(QUser.user.email)).is(email);
-
         return queryOne(List.of(emailCriteria), aclPermission);
     }
 
+    @Override
+    public Mono<User> findByCaseInsensitiveEmail(String email) {
+        String findEmailRegex = String.format("^%s$", email);
+        Criteria emailCriteria = where(fieldName(QUser.user.email)).regex(findEmailRegex, "i");
```

Comment:
I think we need to escape regex special characters in the email before using it as the pattern. Otherwise, if the email is `a.c@gmail.com`, it will match both `a.c@gmail.com` and `abc@gmail.com`. Something like this might do:

```suggestion
        String findEmailRegex = String.format("^%s$", email.replaceAll("[.+*?\\[\\]|]", "\\$0"));
        Criteria emailCriteria = where(fieldName(QUser.user.email)).regex(findEmailRegex, "i");
```

Reply:
@sharat87 Great finding. There was a bug. I've fixed it using the `Pattern.quote` method. Also added a test for it.

Reply:
Ah nice! I didn't realize MongoDB's regex engine supported the `\Q` and `\E` directives that `Pattern.quote` adds.